warp10/src/main/java/io/warp10/continuum/egress/EgressExecHandler.java

```diff
@@ -321,12 +449,23 @@ public void handle(String target, Request baseRequest, HttpServletRequest req, H
         times.add(end - nano);
       }
 
+      if (forcedMacro) {
+        stack.macroClose();
+      }
+
       //
       // Make sure stack is balanced
       //
 
       stack.checkBalanced();
 
+      if (maxtime > 0) {
+        stack.push(maxtime);
+        TIMEBOX.apply(stack);
+        // The call to TIMEBOX above will signal the stack, so we need to handle the signal received.
```

Comment:
When the stack is signaled, an exception is thrown, so the statement below will never be able to handle the signal.

Reply:
The exception is thrown in handleSignal, the call to signal simply sets flags.